### PR TITLE
Aura + CLI

### LINKED ISSUE #44: Control Cavalier using command-line

### PR DESCRIPTION
**Pull request by fsobolev**
Commits got messed on another branch so I created new one and copied everything in one commit.

There are still things to do, so it's a draft for now.

Notes:
* CLI option that accept boolean value require explicit value, just `-f` will not work for example, needs `-f true`. It is possible to make "flags" using CommandLine, but they work as 2 state switches: true or false, with one of them being default. Here we actually need 3 states: do not change, change to true and change to false, that's why explicit value is required.
* I use separate `CmdOptions` instead of `Configuration` because: a) options should be nullable so I can check whether they were used in command or not; b) not all configuration values should be exposed as command; c) some options have different types/values (like roundness which is 0-100 int in command but 0.0-1.0 float in configuration).

Closes #44

### CODE REVIEW

File: NickvisionCavalier.GNOME/NickvisionCavalier.GNOME.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
   <ItemGroup>
     <PackageReference Include="GirCore.Adw-1" Version="0.4.0" />
+    <PackageReference Include="Nickvision.Aura" Version="2023.7.0" />
```
Comment (nlogozzo) [inline NickvisionCavalier.GNOME/NickvisionCavalier.GNOME.csproj:12]: nuget-sources needs update :)

File: NickvisionCavalier.Shared/Models/CmdOptions.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    /// </summary>
+    [Option('o', "drawing-direction", Required = false, HelpText = "Drawing direction, one of: topbottom, bottomtop, leftright, rightleft")]
+    public DrawingDirection? Direction { get; set; }
+    /// <summary>
+    /// The size of spaces between elements
+    /// </summary>
+    [Option('t', "items-offset", Required = false, HelpText = "Offset between items (0-20)")]
+    public uint? ItemsOffset { get; set; }
+    /// <summary>
+    /// Roundness of items (0 - square, 1 - round)
```
Comment (DaPigGuy) [inline NickvisionCavalier.Shared/Models/CmdOptions.cs:50]:
```suggestion
    /// Roundness of items (0 - square, 100 - round)
```

Looks like this should be 100?

File: NickvisionCavalier.Shared/Models/CmdOptions.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    public bool? Filling { get; set; }
+    /// <summary>
+    /// Thickness of lines when filling is off (in pixels)
+    /// </summary>
+    [Option('l', "lines-thickness", Required = false, HelpText = "Lines thickness when filling is off (0-100)")]
+    public uint? LinesThickness { get; set; }
+    /// <summary>
+    /// Active drawing mode
+    /// </summary>
+    [Option('d', "drawing-mode", Required = false, HelpText = "Drawing more, one of: wavebox, levelsbox, particlesbox, spinebox, barsbox")]
```
Comment (DaPigGuy) [inline NickvisionCavalier.Shared/Models/CmdOptions.cs:67]:
```suggestion
    [Option('d', "drawing-mode", Required = false, HelpText = "Drawing mode, one of: wavebox, levelsbox, particlesbox, spinebox, barsbox")]
```

File: NickvisionCavalier.Shared/Models/CmdOptions.cs (hunk 2 above)
Comment (fsobolev) [reply NickvisionCavalier.Shared/Models/CmdOptions.cs:50]: right, here it's 100

**Comment by fsobolev** [comment]
If the user starts the first instance of Cavalier with `--help` option or with any option that contains error, help will be printed to console and the app will not start. If the same is done while Cavalier is already running, this dialog will appear.

image

**Comment by fsobolev** [comment]
Rarely, but still sometimes CAVA wasn't stopping together with the app, latest commit will hopefully fix it. From the docs:

> The [Kill](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill?view=net-7.0) method executes asynchronously. After calling the Kill method, call the [WaitForExit](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexit?view=net-7.0) method to wait for the process to exit

I'm going to update changelog and then the PR is ready for review. Depends on https://github.com/NickvisionApps/Aura/pull/4 but other than saving config everything works.

File: NickvisionCavalier.Shared/Controllers/MainWindowController.cs [hunk 4]
```diff
@@ -45,18 +44,54 @@ public class MainWindowController
     /// </summary>
     public bool AutohideHeader => Configuration.Current.AutohideHeader;
 
+    private readonly PreferencesViewController _preferencesViewController;
```
Comment (nlogozzo) [inline NickvisionCavalier.Shared/Controllers/MainWindowController.cs:47]: Turn this into a public get, init property.

```
public PreferencesViewController PreferencesViewController { get; init; }
```

Then you can also remove the `GetPreferencesViewController()` method.

**Review by nlogozzo — CHANGES_REQUESTED** [review]

**Comment by fsobolev** [comment]
> The scroll window inside a MessageDialog looks weird IMO...could you make it like the style of the CorruptedFilesDialog/HistoryDialog?

image

**Comment by nlogozzo** [comment]
> > The scroll window inside a MessageDialog looks weird IMO...could you make it like the style of the CorruptedFilesDialog/HistoryDialog?
> 
> image
> 

Beautiful! Since there's no description however, maybe we don't need to have the scroll window with the `view` background...

**Comment by fsobolev** [comment]
> Since there's no description however, maybe we don't need to have the scroll window with the view background...

But we still have heading and empty headerbar that don't scroll and should not. IMO this dialog template works good even without description.

**Review by nlogozzo — APPROVED** [review]

**Pull request merged**